src/preferences/upgrade.cpp

```diff
@@ -433,9 +434,12 @@ UserSettingsPointer Upgrade::versionUpgrade(const QString& settingsPath) {
         }
     }
 
-    if (configVersion.startsWith("1.12") ||
-        configVersion.startsWith("2.0") ||
-        configVersion.startsWith("2.1.0")) {
+    QStringList versionParts = configVersion.split(".");
+    QVersionNumber configFileVersion(versionParts[0].toInt(), versionParts[1].toInt(), versionParts[2].toInt());
```

Comment:
Thank you, this looks very good, minor improvement in `const` new API:
```suggestion
#if QT_VERSION < QT_VERSION_CHECK(6, 4, 0)
    const QStringList versionParts = configVersion.split(".");
    const QVersionNumber configFileVersion(versionParts[0].toInt(), versionParts[1].toInt(), versionParts[2].toInt());
#else
    const auto configFileVersion = QVersionNumber::fromString(configVersion);
#endif
```